### ISSUE TITLE
No disk space and writeDataToDestination method

### ISSUE THREAD

**Issue by andre-pelisoli** [smanikandan14/ThinDownloadManager#66]
I faced the situation where the available disk space is over when downloading a file and exceptions occur and were handled within the writeDataToDestination method.

The problem is that as there is no disk space, these exceptions occur continuously and the only way to return from writeDataToDestination method is to write successfully bytes in the file, but it will not happen.

I think that could be adopted other approaches to prevent this. For example, we could have a number of data writing attempts and if these attempts are exhausted, the method returns a result indicating that a problem has occurred.

What do you think guys?

**Comment by smanikandan14** [comment]
@andre-pelisoli Addressed the issue. https://github.com/smanikandan14/ThinDownloadManager/commit/4348d4c77cdae8ec7e7892a0f79d2e52c3376249 Thanks for pointing out. The fix is available in version ```1.2.5```. Let me know how it goes for you with updated version.

**Comment by andre-pelisoli** [comment]
@smanikandan14 Sorry to take a long time to give you a feedback.

I did some tests using the version 1.2.5 and I noticed when an error occurs in writeDataToDestination the download process does not stop. It is a problem because transferData method continues reading bytes from inputstream and at the end we get a corrupted file.

I made a pull request (#75) where I finish the download process with failed status when some error occurs in writeDataToDestination method. Take a look and check if its a reasonable fix to the project.
